pkg/terraform/exec.go

```diff
@@ -0,0 +1,106 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+	"github.com/rs/zerolog/log"
+)
+
+const terraformPath = "terraform"
+
+func CheckTfFmt(dir string) (bool, string) {
+	ok, output, tf := tfInit(dir)
+	if !ok {
+		return ok, output
+	}
+
+	return tfFormat(tf)
+}
+
+func CheckTfValidate(dir string) (bool, string) {
+
+	ok, output, tf := tfInit(dir)
+	if !ok {
+		return ok, output
+	}
+
+	return tfValidate(tf)
+}
+
+func CheckTfLint(dir string) (bool, string) {
+
+	ok, output, tf := tfInit(dir)
+	if !ok {
+		return ok, output
+	}
+
+	return tfValidate(tf)
+}
+
+func tfInit(dir string) (bool, string, *tfexec.Terraform) {
+	workingDir := dir
+	tf, err := tfexec.NewTerraform(workingDir, terraformPath)
+	if err != nil {
+		log.Error().Err(err).Msg("error creating Terraform object")
+		return false, "", nil
+	}
+
+	// TODO: do we need init upgrade ? and init backend ?
+	err = tf.Init(context.Background(), tfexec.Upgrade(true), tfexec.Backend(false))
+	if err != nil {
+		log.Error().Err(err).Msg("error running terraform init")
+		return false, err.Error(), nil
+	}
+
+	return true, "", tf
+}
+
+func tfValidate(tf *tfexec.Terraform) (bool, string) {
+	validationError, err := tf.Validate(context.Background())
+	if err != nil {
+		log.Error().Err(err).Msg("error running terraform validate")
+		return false, "Error running `terraform validate` " + err.Error()
+	}
+	if validationError.ErrorCount != 0 {
+		errorString := ""
+		for _, diag := range validationError.Diagnostics {
+			errorString += "`" + diag.Range.Filename + "` line:`" + strconv.Itoa(diag.Range.Start.Line) + "`"
+			errorString += "\n"
+			errorString += diag.Summary
+			errorString += "\n"
+			errorString += diag.Detail
+			errorString += "\n"
+			errorString += "\n"
```

Comment:
we could create a package to format stuff in Markdown with common function like this